network-formula/network/wicked/files/usr/local/sbin/saltsafe_if

```diff
@@ -0,0 +1,286 @@
+#!/bin/bash
+# ifup/ifdown wrapper script ensuring safe operation if called remotely through Salt
+# Copyright (C) 2023 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+set -Cu
+
+extra="${2:-none}"
+
+base='/etc/sysconfig/network'
+base_backup="$base/salt-backup"
+
+self="$(basename $0)"
+call="${self##*_}"
+
+logtool="$(command -v logger) -t saltsafe" || logtool=echo
+
+
+fail() {
+	echo "$1"
+	exit 1
+}
+
+if [ ! \( "$call" == 'ifdown' -o "$call" == 'ifup' -o "$call" == 'ifroutes' \) ]
+then
+	fail 'Invalid action. Call this script as `saltsafe_ifup` or `saltsafe_ifdown` or `saltsafe_ifroutes`.'
+fi
+
+if ! command -v wicked >/dev/null
+then
+	fail 'Tool requires wicked.'
+fi
+
+log() {
+	local msg="$2"
+	case $1 in
+		0 )
+			$logtool "$msg"
+			if [ "$logtool" != 'echo' ]
+			then
+				echo "$msg"
+			fi ;;
+		1 )
+			if [ "$logtool" == 'echo' ]
+			then
+				local msg="saltsafe: $msg"
+				>&2 $logtool "$msg"
+			else
+				$logtool -s "$msg"
+			fi ;;
+		* )
+			fail 'Invalid function call' ;;
+	esac
+
+}
+
+quit() {
+	case "$1" in
+		0 ) result="$result result=True" ;;
+		1 ) result="$result result=False" ;;
+	esac
+	echo
+	log 0 "$result"
+	exit "$1"
+}
+
+check() {
+	if ! ping -c3 -w5 -q "$master_ip" >/dev/null
+	then
+		return 1
+	fi
+	if ! timeout 20 salt-call -t15 --out quiet test.ping
+	then
+		return 1
+	fi
+}
+
+rollback() {
+	rollback=yes
+	cp -v "$file_backup" "$file"
+}
+
+run() {
+	if [ "$call" == 'ifroutes' ]
+	then
+		local call='systemctl reload network'
+	else
+		local call="$call $interface"
+	fi
+	timeout --preserve-status -k 60 30 $call
+}
+
+backup() {
+	if [ -f "$file_backup" ] && command -v old >/dev/null
+	then
+		old "$file_backup" >/dev/null
+	fi
+	cp "$file" "$file_backup"
+}
+
+run_test() {
+	if [ "$call" == 'ifroutes' ]
+	then
+		#desired_routes=$(awk '{ print $1 "_" $2 | "sort -u" }' "$file")
+		desired_routes="$(echo $routes | tr ',' '\n' | sort)"
+		existing_routes=$({ ip -br -4 r; ip -br -6 r; } | sort -u | awk '!/fe80|::1|scope link/{ print $1 "_" $3 }')
```

Comment:
The awk call match `fe80` and `::1` anywhere in the line, which means it will also filter out things like `ab:fe80`.

I'd recommend to use `^` in the regex.

Reply:
Oh, very good catch!

Reply:
Amended, mind double checking again?

Reply:
looks good :-)

Reply:
Thanks!